### ISSUE TITLE
Default to an unfinished task when creating a new line after a task

### ISSUE THREAD

**Issue by michaelr** [fabiospampinato/vscode-todo-plus#187]
This would make it faster to create new tasks.

I'm i'm typing a task like this:
☐ task

Then press I enter, the next line should default to a task like this:
☐ task
☐ <cursor is here>

**Comment by fabiospampinato** [comment]
Sounds like a good addition, I'm not sure if I'd want to use that though 🤔 It may complicate things a little when you cant to add a new todo indented differently.

I'd be happy to consider a PR that adds an option for implementing this 👍

**Comment by LuisARodr** [comment]
I'll be taking a look at this issue.

**Comment by OiYouYeahYou** [comment]
Coming from word processors and good markdown editors, I expect when on an existing todo and hitting enter another todo is added, like bullet points. Implementation wise, I would be happy with an opt-in setting for this behaviour as most I've spoken to haven't understood why this is a preference.

Here's my expected behaviour:

 * `Single enter` add a new todo
 * `Double enter` or `Enter on a blank todo` remove added todo and begin para
 * `Shift enter` don't add todo and remain at previous task indent
 * Possibly `on save` remove blank todos

**Comment by fabiospampinato** [comment]
@OiYouYeahYou feel free to submit a PR implementing that. I'd leave out the automatic formatting/cleanup on save though, as it seems out of scope.

**Comment by LuisARodr** [comment]
@fabiospampinato @OiYouYeahYou   I'm working on those features, I'm going to open a pull request of the issue as a WIP.

**Comment by LuisARodr** [comment]
Currently working on the PR, here is my progress so far:

Current features:

- Added new configuration `todo.automaticSymbol`

    ```
    "todo.automaticSymbol": false // Automatically insert new todo symbols and remove them when pressing enter"
    ```
- When an enter is pressed and you're currently on an empty todo; it deletes the empty todo.
- When an enter is pressed and you're currently on an filled todo; it creates an empty todo next line.

Missing:
- Using shift+enter to make a new line from a filled todo without creating an empty todo.

Any changes or suggesting are accepted

**Comment by michaelr** [comment]
Are you using the LanguageConfiguration onEnterRules to do this?
https://code.visualstudio.com/api/references/vscode-api#LanguageConfiguration

**Comment by LuisARodr** [comment]
No, I'm using `onDidChangeTextDocument` to catch a new line change, looking at `onEnterRules` it seems like it might be possible to use it for inserting a todo when an enter is pressed, but I don't see how it can remove a empty todo, or help to recognize a `Shift+Enter` without keybinding a command.

**Comment by michaelr** [comment]
if you bind shift+enter and call editor.action.insertLineAfter it seems to
preserve the indentation and move the cursor.

**Comment by LuisARodr** [comment]
Currently on the [PR#189](https://github.com/fabiospampinato/vscode-todo-plus/pull/189) there is a functioning version of the feature, there is still a small issue I'll be taking a look.

**Comment by LuisARodr** [comment]
@fabiospampinato  Just finished the [PR](https://github.com/fabiospampinato/vscode-todo-plus/pull/189) for the issue, if there are any changes that need to be done I'll come back to do them.

**Comment by fabiospampinato** [comment]
@LuisARodr Thanks, I'll take a look at that as soon as I can :+1:

**Comment by fabiospampinato** [comment]
@LuisARodr sorry for the delay, I haven't forgotten about this, let me get [Notable v1.5](https://github.com/notable/notable) ready (a few more issues to address) and I'll get to this 👍
